modules/acm/src/main/java/edu/gemini/epics/acm/CaCommandMonitorImpl.java

```diff
@@ -106,9 +143,12 @@ public synchronized void completeSuccess() {
         if (currentState == State.BUSY) {
             currentState = State.IDLE;
             condition.signalAll();
-            lock.unlock();
-            if (callback != null) {
-                callback.onSuccess();
+            try {
+                if (callback != null) {
+                    callback.onSuccess();
+                }
+            } finally {
+                lock.unlock();
```

Comment:
The `unlock` was moved after the callback call, because otherwise it was possible for code calling `waitDone` to not see the effect of the callback. Same for `competeFailure` and `completePause`.